resolwe_bio/tools/make_report.py

```diff
@@ -0,0 +1,154 @@
+    return table_data, header
+
+
+def parse_targetPCRmetrics(metrics_report):
+    """Parse CollectTargetedPcrMetrics report file."""
+    with open(metrics_report) as pcr_metrics:
+        labels = []
+
+        for line in pcr_metrics:
+            if line.startswith('#'):
+                continue
+            if len(labels) == 0:
+                if line.startswith('CUSTOM_AMPLICON_SET'):
+                    labels = line.strip().split('\t')
+            else:
+                values = line.strip().split('\t')
+                break
+
+        return dict(zip(labels, values))
+
+
+def coverage_uniformity(covd, mean_coverage, cutoff):
+    """Parse coverageBed (.covd) report file."""
+    with open(covd) as pbc:
+        covered_bases = 0
+        all_bases = 0
+        for line in pbc:
+            cov_data = line.strip().split('\t')
+            if int(cov_data[6]) >= mean_coverage * cutoff:
+                covered_bases = covered_bases + 1
+            all_bases = all_bases + 1
+    return float(covered_bases) / float(all_bases)
+
+
+def make_barplot(x, y, savefile=None, dpi=300):
+    """Make barplot."""
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+
+    ax.bar(x, y, 0.5, color=GENIALIS_BLUE, linewidth=0.0)
+    ax.set_xlim(min(x), max(x))
+    ax.set_ylim(0, 1.1)
+    ax.set_xlabel(r'Amplicon')
+    ax.set_ylabel('Per amplicon coverage')
+
+    plt.savefig(savefile, dpi=dpi)
+
+
+def make_scatterplot(x, y, hline=None, savefile=None, dpi=300):
+    """Make scatterplot."""
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+
+    ax.plot(x, y, color=GENIALIS_BLUE, marker='o', linestyle=' ', markeredgewidth=0.0)
+    ax.set_xlim(min(x), max(x))
+    ax.set_yscale('log')
+    ax.set_xlabel(r'Amplicon')
+    ax.set_ylabel('Log total coverage')
+    if hline:
+        ax.axhline(hline, color=GENIALIS_GREY_2)
+
+    plt.savefig(savefile, dpi=dpi)
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+
+    # Open template ind fill it with data:
+    with open(args.template, 'r') as template_in, open('report.tex', 'wt') as template_out:
+        template = template_in.read()
+        template = template.replace('{#LOGO#}', args.logo)
+        template = template.replace('{#SAMPLE_NAME#}', _format_latex(args.sample))
+
+        # Parse metrics file:
+        metrics = parse_targetPCRmetrics(args.metrics)
+        template = template.replace('{#ALIGNED_READS#}', metrics['PF_UQ_READS_ALIGNED'])
+        template = template.replace('{#BASES_ALIGNED#}', str(round(float(metrics['PCT_AMPLIFIED_BASES']), DECIMALS)))
+        template = template.replace('{#BASES_TARGET#}', ' ')
+        template = template.replace('{#COV_MEAN#}', str(round(float(metrics['MEAN_TARGET_COVERAGE']), DECIMALS)))
+        mean20 = round(0.2 * float(metrics['MEAN_TARGET_COVERAGE']), DECIMALS)
+        template = template.replace('{#COV_20#}', str(mean20))
+        template = template.replace('{#COV_UNI#}', str(round(coverage_uniformity(
```

Comment:
Maybe put `str(round(coverage_uniformity(args.covd, float(metrics['MEAN_TARGET_COVERAGE']), 0.2), DECIMALS))` in a new variable (similar to `mean20`) to make the code more readable.